pkg/controllers/managementuser/settings/controller.go

```diff
@@ -42,7 +45,7 @@ func (h *handler) onChange(key string, obj *v3.Setting) (runtime.Object, error)
 		return nil, nil
 	}
 
-	return obj, h.apply.
+	err := h.apply.
```

Comment:
This apply should use a cache so we don't have to hit the downstream API i.e. `WithCacheTypes`